### PR TITLE
Feature: nvidia.com/gpu resource requests

### PR DESCRIPTION
**Pull request by 5had3z**
Remade PR from https://github.com/volcano-sh/volcano/pull/1892 since git was being a nightmare when I was trying to clean up my mistakes, hopefully the git side of this work this is now in an acceptable state.

DESCRIPTION FROM https://github.com/volcano-sh/volcano/pull/1892: 
This is based off peiniliu's work but utilises Nvidia's GPU operator as it does the hard work of annotating nodes with all the details we would ever want. Integrating their label system into volcano increases convenience as it removes the requirement of an additional device plugin and ensures pods are restricted to their scheduled gpu, rather than using environment variables. Other advantage include using nodeSelector in the pod spec with the lablels nvidia provides such as nvidia.com/gpu.product to select a particular gpu node.

I've been running this setup on my small kubeadm cluster at my university with no issues so far (4x nodes with 2x 3090 each).

### CODE REVIEW

**Comment by k82cn** [comment]
@5had3z , would you help to create a seperate PR for design doc ?

**Comment by l1n** [comment]
Anything left to do to merge this? It seems like a straightforward improvement on current state.

**Comment by 5had3z** [comment]
> Anything left to do to merge this? It seems like a straightforward improvement on current state.

Not unless there's some change to the scope of things, I've been using it at my cluster at uni for a few months now.

**Comment by l1n** [comment]
Do you have a chart for deploying your branch of this that could be tested?

**Comment by 5had3z** [comment]
> Do you have a chart for deploying your branch of this that could be tested?

I just changed the deployment.yaml to use my own images on a private registry, I can certainly push them to docker hub and you can just change the deployment.yaml to point to mine. I'll comment here again with the hub addresses when it's done (maybe a day or so).

**Comment by 5had3z** [comment]
Images vc-scheduler/vc-controller-manager/vc-webhook-manager are on my docker hub now (5had3z)

File: pkg/scheduler/api/node_info.go [hunk 1]
```diff
@@ -281,23 +281,30 @@ func (ni *NodeInfo) setNodeGPUInfo(node *v1.Node) {
...
 		return
 	}
-	memory, ok := node.Status.Capacity[VolcanoGPUResource]
-	if !ok {
-		return
-	}
-	totalMemory := memory.Value()
 
-	res, ok := node.Status.Capacity[VolcanoGPUNumber]
+	res, ok := node.Status.Capacity[GPUResourceName]
```
Comment (wpeng102) [inline pkg/scheduler/api/node_info.go:285]: https://github.com/volcano-sh/devices/ will use `VolcanoGPUNumber` as resource name. If change to `GPUResourceName` how to work with volcano-sh/devices

**Pull request closed without merging**